syntax = "proto3";

import "mediocre/image/v1beta/image.proto";;
import "mediocre/options/v1beta/options.proto";

package mediocre.image.rotate.v1beta;

service RotateService {
  rpc Rotate(RotateRequest) returns (RotateResponse);
}

message RotateRequest {
  image.v1beta.Image image = 1;
  RotateParams params = 2;
}

message RotateParams {
  double angle = 1 [(options.v1beta.mediocre_field_options).min = 0, (options.v1beta.mediocre_field_options).max = 360];
}

message RotateResponse {
  image.v1beta.Image image = 1;
}
